app.ts

```diff
@@ -0,0 +1,388 @@
+			if (endID === count) {
+				startID = ((globals.currentPage - 1) * maxRecords) + 1;
+				endID = count;
+				globals.firstPage = pageStart;
+				globals.lastPage = pageEnd;
+			}
+		})
+		.then(() => {
+			return pageNumbers(pageStart, pageEnd);
+		})
+		.then(() => {
+			return showRecords(startID, endID);
+		})
+		.catch((error) => {
+			throw error;
+		})
+}
+
+// When adjusting the height and on different screen sizes. This function would responsible for calculating how much records should be displayed based on the height of the window itself. 
+function adjustDisplayedRecords() {
+	if (isFunctionRunning) {
+		return;
+	}
+	isFunctionRunning = true;
+	let screenHeight = $(window).height() as number;
+	if (screenHeight < 68) {
+		screenHeight = 68;
+	}
+	let maxRecords: number;
+	let pageStart: number;
+	let pageEnd: number;
+	return apiManager.getRecordCount()
+		.then((count) => {
+			let maxRecords = Math.floor(parseInt(screenHeight as any) / 68);
+			let inputNumber = input();
+			let length = inputNumber.length as number;
+			let inputNumberInt = parseInt(inputNumber);
+			if (inputNumber === '') {
+				let newCurrentPage = Math.ceil(globals.currentFromID / maxRecords);
+				if (newCurrentPage === 1) {
+					globals.currentFromID = 1;
+				}
+				globals.currentToID = globals.currentFromID + (maxRecords - 1);
+				globals.currentPage = newCurrentPage;
+				let originalID = (Math.floor(maxRecords * globals.currentPage) - (maxRecords - 1));
+				globals.difference = globals.currentFromID - originalID;
+			} else {
+				if (length > 0) {
+					let newCurrentPage = Math.ceil(inputNumberInt / maxRecords);
+					if (globals.currentToID > count) {
+						globals.currentToID = count;
+					}
+					globals.currentToID = newCurrentPage * maxRecords;
+					globals.currentPage = newCurrentPage;
+					globals.currentFromID = (globals.currentPage - 1) * maxRecords + 1;
+				}
+			};
+			pageEnd = Math.ceil(Math.floor(globals.currentToID / maxRecords) / 10) * 10;
+			pageStart = pageEnd - 9;
+			globals.firstPage = pageStart;
+			globals.lastPage = pageEnd;
+			$("tbody").empty();
+			isFunctionRunning = false;
+			return showRecords(globals.currentFromID, globals.currentToID);
+		})
+		.then(() => {
+			return pageNumbers(pageStart, pageEnd);
+		})
+		.then(() => {
+			return maxRecords;
+		})
+		.catch((error) => {
+			throw error;
+		})
+}
+
+// Calls the function to resize with a timeout added for precision
+let resizeTimer: ReturnType<typeof setTimeout>;
+function resize() {
+	clearTimeout(resizeTimer);
+	resizeTimer = setTimeout(() => {
+		adjustDisplayedRecords();
+	}, 250);
+}
+
+// Just a loader to display when the table is empty and records is being fetched. 
+function loader() {
+	let content = $('tbody').text();
+	if (content === '') {
+		$('.results').append('<div class="loader"></div>');
+	} else {
+		$('.loader').css({ 'display': 'none' });
+	}
+}
+
+// Calculate how many records should be displayed on the screen height
```

Comment:
Similar comment about javadoc style. Except what you have written is pretty much already correct, I would just change `Calculate` to `Calculates`